week3/src/components/Card.jsx

```diff
@@ -0,0 +1,84 @@
+import styled, { css, keyframes } from "styled-components";
+import front from "../assets/img/clover.jpg";
+
+const Card = ({ id, src, alt, flipped, onClick }) => {
+  return (
+    <CardWrapper onClick={() => onClick(id)}>
+      <CardFlipper flipped={flipped ? "true" : undefined}>
+        <CardFront src={front} alt="카드 앞면" />
+        <CardBack src={src} alt={alt} />
+      </CardFlipper>
+    </CardWrapper>
+  );
+};
+
+export default Card;
+
+// 애니메이션
+const flipAnimation = keyframes`
+    from {
+        transform: rotateY(0deg);
+    }
+    to {
+        transform: rotateY(180deg);
+    }
+`;
+
+const unflipAnimation = keyframes`
+    from {
+        transform: rotateY(180deg);
+    }
+    to {
+        transform: rotateY(360deg);
+    }
+`;
+
+const CardWrapper = styled.article`
+  width: 10rem;
+  height: 15rem;
+`;
+
+const CardFlipper = styled.div`
+  width: 100%;
+  height: 100%;
+  position: relative;
+  object-fit: cover;
+  transform-style: preserve-3d;
+  transition: transform 0.6s;
+  ${({ flipped }) =>
+    flipped &&
+    css`
+      transform: rotateY(180deg);
+    `}
+  ${({ flipping }) =>
+    flipping &&
+    css`
+      animation: ${flipAnimation} 0.6s forwards;
+    `}
+    ${({ unflipping }) =>
+    unflipping &&
+    css`
+      animation: ${unflipAnimation} 0.6s forwards;
+    `}
```

Comment:
여기 내부에서 css를 이용해서 템플릿 리터럴을 사용했는데 
```suggestion
const StyledComponent = styled.div`
  transform: ${({ flipped }) => flipped && 'rotateY(180deg)'};
  animation: ${({ flipping }) => flipping && `${flipAnimation} 0.6s forwards`};
`
```
이렇게 css를 사용하지 않고 작성할 수 있답니다!